resource-managers/kubernetes/core/src/test/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocatorSuite.scala

```diff
@@ -655,7 +656,9 @@ class ExecutorPodsAllocatorSuite extends SparkFunSuite with BeforeAndAfter {
 
   test("print the pod name instead of Some(name) if pod is absent") {
     val nonexistentPod = "i-do-not-exist"
-    val conf = new SparkConf().set(KUBERNETES_DRIVER_POD_NAME, nonexistentPod)
+    val conf = new SparkConf()
+      .set(KUBERNETES_DRIVER_POD_NAME, nonexistentPod)
+      .set(KUBERNETES_DRIVER_SUBMIT_CHECK, true)
```

Comment:
If we need to enforce this to the users, this is a regression.

Reply:
It's not a regression. We only need to do this unit test. KUBERNETES_DRIVER_SUBMIT_CHECK is an internal one and should be set by Spark only